src/mgr/Gil.cc

```diff
@@ -14,11 +14,29 @@
 
 
 #include "Python.h"
-
+#include "mgr/mgr_perf_counters.h"
+#include "common/BackTrace.h"
 #include "common/debug.h"
+#include <atomic>
+
+static std::atomic<PyThreadState *> g_last_gil_ts { 0 };
```

Comment:
Well, this won't be completely accurate because we acquire the GIL in many places in python code that won't be tracked:

- returns from syscalls
- returns from python bindings like cephfs or rados

So we should include some comment here communicating that.

Reply:
@batrick sorry, I didn't get it, we are just recording the thread state, so we can know which last thread state is holding the current gil and we are waiting for.
But now I think it is not a good solution for contention when the gil was not released. we won't have any message indicating we are waiting to X.. maybe add an outside alarm that will pop after 1 minute of waiting?